### PR TITLE
Add script for building ommpfritt with icons (original by pasbi)

### PR DESCRIPTION
**Pull request by mestaritonttu**
Now prints usage instructions, if no argument given.

### CODE REVIEW

File: build-scripts/build-omm-with-icons.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+if [ ! "$1" ]; then
+    echo "$usage"
+    exit 1
+fi
+
+set -e
+
+source=$1
+shift
+config=${1:-Release}
```
Comment (pasbi) [inline build-scripts/build-omm-with-icons.sh:17]: If the configuration argument is not set, but additional arguments are passed to cmake  (`"$@"` in line 31), this shift will eat the first of those arguments.

Proper solution would be named arguments (e.g. `--config Release`) or non-optional argument .

File: build-scripts/build-omm-with-icons.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Release
+Debug
+
+You can also pass additional arguments to CMake.
+"
+
+if [ ! "$2" ]; then
+    echo "$usage"
+    exit 1
+fi
```
Comment (pasbi) [inline build-scripts/build-omm-with-icons.sh:17]: How about making this more idiomatic?

```suggestion
usage="Usage: $0 SOURCE CONFIG [CMAKE_ARGUMENTS ...]

This script builds ommpfritt with icons from SOURCE (root of this repository).
CONFIG can be 'Release' or 'Debug'.
Builds and installs the application into '/tmp', see output for details.
Additional CMAKE_ARGUMENTS are passed verbatim to the cmake configuration call.
"

if [[ "$1" =~ (-h|--help) ]]; then
    echo "$usage"
    exit 0
fi

if [ ! "$2" ]; then
    echo "$usage"
    exit 1
fi
```

**Review by pasbi — APPROVED** [review]

**Comment by pasbi** [comment]
Somehow the actions are not triggered.
I suspect that is because you're PR comes from a fork.
I have checked the settings and it says "Require approval for first-time contributors who are new to GitHub" (the least restrictive option).
I have no clue if you are considered a "first-time contributor" or "new to GitHub" or how I can approve you (I have approved your changes, but that didn't change anything).

**Comment by mestaritonttu** [comment]
> Somehow the actions are not triggered. I suspect that is because you're PR comes from a fork. I have checked the settings and it says "Require approval for first-time contributors who are new to GitHub" (the least restrictive option). I have no clue if you are considered a "first-time contributor" or "new to GitHub" or how I can approve you (I have approved your changes, but that didn't change anything).

Seems like the same crap as in #201. Don't all PRs come from a fork? That's how GitHub was designed. Actions is simply a broken feature.

**Comment by pasbi** [comment]
No, I myself also contribute via PRs which originate from the base repo, not from a fork.
Actions are quite useful canary to check code quality.
But it does not make sense at all that the actions can't run on forks

**Comment by mestaritonttu** [comment]
> Ah, now it makes sense to me! The actions script contained `on: [push]`, I've changed those lines to `on: [push, pull_request]` and now it seems to work.

Great that you solved it! I think now the contribution workflow is at a good stage and I should start promoting the project.

**Pull request merged**